samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -0,0 +1,143 @@
+"""
+Remote invoke executor implementation for Lambda
+"""
+import base64
+import json
+import logging
+from json import JSONDecodeError
+from typing import Any, Dict, cast
+
+from botocore.response import StreamingBody
+
+from samcli.lib.remote_invoke.exceptions import InvalidResourceBotoParameterException
+from samcli.lib.remote_invoke.remote_invoke_executors import (
+    BotoActionExecutor,
+    RemoteInvokeExecutionInfo,
+    RemoteInvokeRequestResponseMapper,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class LambdaInvokeExecutor(BotoActionExecutor):
+    """
+    Calls "invoke" method of "lambda" service with given input.
+    If a file location provided, the file handle will be passed as Payload object
+    """
+
+    _lambda_client: Any
+    _function_name: str
+    _SUPPORTED_PARAMETERS: dict = {
+        "InvocationType": ["Event", "RequestResponse", "DryRun"],
+        "LogType": ["None", "Tail"],
+        "Qualifier": [],
+    }
+
+    def __init__(self, lambda_client: Any, function_name: str):
+        self._lambda_client = lambda_client
+        self._function_name = function_name
+        self.request_parameters = {"InvocationType": "RequestResponse", "LogType": "Tail"}
+
+    def validate_action_parameters(self, parameters: dict):
+        """
+        Validates the input boto parameters and prepares the parameters for calling the API.
+
+        :param parameters: Boto parameters provided as input
+        """
+        for parameter_key, parameter_value in parameters.items():
+            parameter_allowed_values = self._SUPPORTED_PARAMETERS.get(parameter_key, [])
+            if parameter_key == "FunctionName":
+                LOG.info("FunctionName is defined using the value provided for --resource-id option.")
+            elif parameter_key == "Payload":
+                LOG.info("Payload is defined using the value provided for either --payload or --payload-file options.")
+            elif parameter_key not in self._SUPPORTED_PARAMETERS:
```

Comment:
I am not sure if we want to have supported parameters here. Can't we just pass these parameters to the boto3 calls, and let boto3 to handle if it is a valid parameter or not?